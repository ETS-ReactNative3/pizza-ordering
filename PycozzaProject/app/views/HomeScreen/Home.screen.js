import React, { Component } from 'react'
import { Text, View, ImageBackground, StyleSheet } from 'react-native';
import color from '../../resources/colors'
import { Button } from 'react-native-elements';
import dimension from '../../resources/dimensions';
export default class HomeScreen extends Component {
    render() {
        return (
            <View style={styles.container}>
                <ImageBackground
                    style={{ width: dimension.window.width, height: dimension.window.height, }}
                    source={
                        require('../../assets/images/home-background.jpg')
                    }
                >

                    <View style={{ alignItems: 'flex-start', marginTop: 'auto', marginBottom: 100, marginLeft: 30, }}>
                        <View style={{marginBottom: 20}}>
                            <Text style={styles.titleStyle}>PYCOZZA</Text>
                            <Text style={styles.subtitleStyle}>Choose your Pizza</Text>
                        </View>
                        <Button title='Start order    >' buttonStyle={{
                            backgroundColor: color.mainColor, borderRadius: 20, width: 160
                        }} titleStyle={{ marginLeft: 20 }} />
                    </View>
                </ImageBackground>

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
    },
    titleStyle: {
        fontSize: 40,
        color: color.textColor
    },
    subtitleStyle: {
        fontSize: 17,
        color: color.textColor

    }
})