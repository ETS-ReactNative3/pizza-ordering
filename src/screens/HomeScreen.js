import React, {Component} from 'react';
import {
  View,
  Text,
  ImageBackground,
  TouchableOpacity,
  StatusBar,
  Image,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import {dimensionStyles} from '../resources/dimension.js';
import {string} from '../resources/string.js';
import {textStyle} from '../resources/textStyle.js';
import HomeImg from '../../assets/HomeImg.jpg';
import Logo from '../../assets/Logo.png';

export default class HomeScreen extends Component {
  navigateToCategoryScreen = () => {
    const {navigation} = this.props;
    navigation.navigate('Category Screen');
  };
  render() {
    return (
      <View style={dimensionStyles.HomeContainer}>
        <StatusBar hidden />
        <ImageBackground
          style={dimensionStyles.HomeImg}
          source={HomeImg}
          resizeMode="cover"
        />
        <Image
          style={dimensionStyles.Logo}
          source={Logo}
          resizeMode="contain"
        />
        <View style={dimensionStyles.HomeNameContainer}>
          <Text style={textStyle.restaurant}>Restaurant</Text>
          <Text style={textStyle.restaurantName}>{string.restaurantName}</Text>
          <TouchableOpacity
            style={dimensionStyles.StartOrderButton}
            onPress={() => this.navigateToCategoryScreen()}>
            <Text style={textStyle.StartOrderButton}>START ORDER</Text>
            <View style={dimensionStyles.ArrowRightIconHome}>
              <Icon name="angle-right" size={20} color="#FFFFFF" />
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
