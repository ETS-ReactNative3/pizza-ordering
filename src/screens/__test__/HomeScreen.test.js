import React from 'react';
import {shallow, mount} from 'enzyme';

import HomeScreen from './../HomeScreen.js';

describe('Home Screen Test', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
  });

  it('should render snapshot test for home screen', () => {
    const wrapper = mount(<HomeScreen />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should navigate to Category Screen', () => {
    let props = {navigation: {navigate: jest.fn()}};
    const wrapper = mount(<HomeScreen {...props} />);
    wrapper.find('TouchableOpacity').simulate('click');
    expect(props.navigation.navigate).toBeCalledWith('Category Screen');
  });
});
