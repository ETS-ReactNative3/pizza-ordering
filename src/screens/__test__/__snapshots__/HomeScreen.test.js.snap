// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Test should render snapshot test for home screen 1`] = `
<HomeScreen>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <StatusBar
        animated={false}
        hidden={true}
        showHideTransition="fade"
      />
      <ImageBackground
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../assets/HomeImg.jpg",
          }
        }
        style={
          Object {
            "height": 1334,
            "position": "absolute",
            "width": 750,
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "height": 1334,
              "position": "absolute",
              "width": 750,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": 1334,
                "position": "absolute",
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../assets/HomeImg.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 1334,
                    "width": 750,
                  },
                  undefined,
                ]
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../assets/HomeImg.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    undefined,
                  ]
                }
              />
            </Image>
          </View>
        </View>
      </ImageBackground>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../assets/Logo.png",
          }
        }
        style={
          Object {
            "height": 100,
            "left": 20,
            "position": "absolute",
            "top": 20,
            "width": 100,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/Logo.png",
            }
          }
          style={
            Object {
              "height": 100,
              "left": 20,
              "position": "absolute",
              "top": 20,
              "width": 100,
            }
          }
        />
      </Image>
      <View
        style={
          Object {
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 47,
                "letterSpacing": 0.1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 47,
                  "letterSpacing": 0.1,
                }
              }
            >
              Restaurant
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
                "letterSpacing": 0.07,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 30,
                  "letterSpacing": 0.07,
                }
              }
            >
              Pizza Pazzi
            </Text>
          </Text>
          <ForwardRef
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#e5293e",
                "borderRadius": 19,
                "height": 38,
                "justifyContent": "center",
                "marginTop": 10,
                "width": 337.5,
              }
            }
          >
            <TouchableOpacity
              hostRef={null}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#e5293e",
                  "borderRadius": 19,
                  "height": 38,
                  "justifyContent": "center",
                  "marginTop": 10,
                  "width": 337.5,
                }
              }
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#e5293e",
                      "borderRadius": 19,
                      "height": 38,
                      "justifyContent": "center",
                      "marginTop": 10,
                      "width": 337.5,
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#e5293e",
                        "borderRadius": 19,
                        "height": 38,
                        "justifyContent": "center",
                        "marginTop": 10,
                        "width": 337.5,
                      },
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#e5293e",
                        "borderRadius": 19,
                        "height": 38,
                        "justifyContent": "center",
                        "marginTop": 10,
                        "opacity": 1,
                        "width": 337.5,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#e5293e",
                          "borderRadius": 19,
                          "height": 38,
                          "justifyContent": "center",
                          "marginTop": 10,
                          "opacity": 1,
                          "width": 337.5,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#FFFFFF",
                            "fontSize": 15,
                            "letterSpacing": 0.07,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "#FFFFFF",
                              "fontSize": 15,
                              "letterSpacing": 0.07,
                            }
                          }
                        >
                          START ORDER
                        </Text>
                      </Text>
                      <View
                        style={
                          Object {
                            "alignSelf": "flex-end",
                            "paddingRight": 15,
                            "position": "absolute",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignSelf": "flex-end",
                              "paddingRight": 15,
                              "position": "absolute",
                            }
                          }
                        >
                          <Icon
                            allowFontScaling={false}
                            color="#FFFFFF"
                            name="angle-right"
                            size={20}
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#FFFFFF",
                                    "fontSize": 20,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "FontAwesome",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontSize": 20,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                            </Text>
                          </Icon>
                        </View>
                      </View>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </View>
      </View>
    </View>
  </View>
</HomeScreen>
`;
